### PR TITLE
Update Non-Functional Requirements (NFRs) for DG

### LINKED ISSUE #33: Update DG: NFRs

### PR DESCRIPTION
**Pull request by kanna-1**
This commit includes updates to the Non-Functional Requirements (NFRs) for the DG component. Additionally, some new NFRs have been added, with more to come in subsequent commits.

Fixes #33 

Need team members to do PR review before merging.

### CODE REVIEW

**Comment by kanna-1** [comment]
Fixes #33 

In this pull request, we address the issue described in #33 by implementing the following changes.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -324,9 +324,11 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. The application should be compatible with the designated operating systems and hardware configurations, as specified in the system requirements.
+2. The application should respond promptly to user inputs, with minimal latency and loading times for data retrieval and processing.
+3. The user interface should be user-friendly and intuitive, designed to optimize the workflow of frontdesk staff who need to complete tasks within 2-3 minutes.
+4. The application should be designed to handle an increasing volume of patient records efficiently without noticeable performance degradation.
+5. Ensure that the application complies with PDPA and healthcare regulations
```
Comment (kanna-1) [inline docs/DeveloperGuide.md:331]: Missing a period at the end of sentence.

Comment (kanna-1) [reply docs/DeveloperGuide.md]: **Fixed**

**Review by hyc17003 — COMMENTED** [review]
I like how it's more specific, lgtm!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -324,9 +324,11 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. The application should be compatible with the designated operating systems and hardware configurations, as specified in the system requirements.
+2. The application should respond promptly to user inputs, with minimal latency and loading times for data retrieval and processing.
+3. The user interface should be user-friendly and intuitive, designed to optimize the workflow of frontdesk staff who need to complete tasks within 2-3 minutes.
```
Comment (WZWren) [inline docs/DeveloperGuide.md:329]: Can we be more specific for this? Perhaps specify that it is intuitive for a fast typer to perform tasks - we can refer to the given constraints [here](https://nus-cs2103-ay2324s1.github.io/website/admin/tp-constraints.html).

**Review by WZWren — COMMENTED** [review]
I think it is overall OK, but we could squeeze in a bit more, since we are given constraints on the project via the tP requirements.

**Pull request merged**